tests/cases/conformance/jsdoc/overloadTag2.ts

```diff
@@ -0,0 +1,40 @@
+// @checkJs: true
+// @allowJs: true
+// @target: esnext
+// @outdir: foo
+// @declaration: true
+// @filename: overloadTag2.js
+// @strict: true
+export class Foo {
+    #a = true ? 1 : "1"
+    #b
+
+    /**
+     * Should not have an implicit any error, because constructor's return type is always implicit
+     * @constructor
+     * @overload
+     * @param {string} a
+     * @param {number} b
+     */
+    /**
+     * @constructor
+     * @overload
+     * @param {number} a
+     */
+    /**
```

Comment:
Does this test need to be written in such a way as to satisfy #52474?

Reply:
Oh, so #52474 is no longer a thing, okay.

Reply:
Can I offer an alternative thought on the syntax that could satisfy #52474 without causing parser issues? What if overload simply allowed you to declare a method signature? For instance:

```js
/**
 * @overload { (inst, klass, members) => object }
 * @overload { (inst, members) => undefined }
 * @param {object} inst
 * @param {function} klass
 * @param {object} members
 * @returns {object|undefined}
 */
function share(inst, klass, members) {}
``` 
This should be both easy to parse, and satisfy the user's needs while allowing for minimally redundant documentation, all in a single comment.